src/t_hash.c

```diff
@@ -788,7 +788,8 @@ GetFieldRes hashTypeGetValue(redisDb *db, robj *o, sds field, unsigned char **vs
         dbDelete(db,keyObj);
         res = GETF_EXPIRED_HASH;
     }
-    signalModifiedKey(NULL, db, keyObj);
+    if (!(hfeFlags & HFE_LAZY_NO_SIGNAL))
+        signalModifiedKey(NULL, db, keyObj);
```

Comment:
If my understanding is correct, we are okay with calling this function once after doing all the changes. I added this flag and calling it once for new commands. Probably, we should consider this for other commands that are using hashTypeGetValue() api as well, e.g. hmget